From 0a78821ce4f346a10675e1bc64522185f8178076 Mon Sep 17 00:00:00 2001
From: Yuvasree Sivasankaran <quic_ysivasan@quicinc.com>
Date: Thu, 30 Nov 2023 21:41:34 +0530
Subject: [PATCH] wifi: ath12k: Update Pktlog filter to capture data frames

In pktlog, data frames are not capturing. The issue was with
pktlog filter configuration where the rx filter doesn't have MO filters.
Add MO filters HTT_RX_MO_MGMT_FILTER_FLAGS0,
HTT_RX_MO_MGMT_FILTER_FLAGS1,HTT_RX_MO_CTRL_FILTER_FLASG2,
HTT_RX_MO_CTRL_FILTER_FLASG3, HTT_RX_MO_DATA_FILTER_FLASG3
in reference with filter config data from firmware.

Signed-off-by: Yuvasree Sivasankaran <quic_ysivasan@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/debugfs.c | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/debugfs.c
+++ b/drivers/net/wireless/ath/ath12k/debugfs.c
@@ -2663,11 +2663,20 @@ static ssize_t ath12k_write_pktlog_filte
 
 	tlv_filter.rx_filter = rx_filter;
 	if (rx_filter) {
-		tlv_filter.pkt_filter_flags0 = HTT_RX_FP_MGMT_FILTER_FLAGS0;
-		tlv_filter.pkt_filter_flags1 = HTT_RX_FP_MGMT_FILTER_FLAGS1;
-		tlv_filter.pkt_filter_flags2 = HTT_RX_FP_CTRL_FILTER_FLASG2;
-		tlv_filter.pkt_filter_flags3 = HTT_RX_FP_CTRL_FILTER_FLASG3 |
-					       HTT_RX_FP_DATA_FILTER_FLASG3;
+		tlv_filter.pkt_filter_flags0 =
+					HTT_RX_FP_MGMT_FILTER_FLAGS0 |
+					HTT_RX_MO_MGMT_FILTER_FLAGS0;
+		tlv_filter.pkt_filter_flags1 =
+					HTT_RX_FP_MGMT_FILTER_FLAGS1 |
+					HTT_RX_MO_MGMT_FILTER_FLAGS1;
+		tlv_filter.pkt_filter_flags2 =
+					HTT_RX_FP_CTRL_FILTER_FLASG2 |
+					HTT_RX_MO_CTRL_FILTER_FLASG2;
+		tlv_filter.pkt_filter_flags3 =
+					HTT_RX_FP_CTRL_FILTER_FLASG3 |
+					HTT_RX_MO_CTRL_FILTER_FLASG3 |
+					HTT_RX_FP_DATA_FILTER_FLASG3 |
+					HTT_RX_MO_DATA_FILTER_FLASG3;
 	}
 
 	ret = ath12k_dp_tx_htt_rx_filter_setup(ab, ring_id, 0,
